### ISSUE TITLE
In which directory should benchmark be run?

### ISSUE THREAD

**Issue by titq2022** [ultralytics/ultralytics#1143]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I would like to run benchmark.
The exported file is output to the current directory, but `[Errno 2] No such file or directory` is displayed.
In which directory should benchmark be run?
(I installed ultralytics with pip install.)

**current directory**
`/workdir`

**my code**
`yolo benchmark model='yolov8m-seg.pt' half=True device='0' imgsz=640`

**trace**
```
Ultralytics YOLOv8.0.45 🚀 Python-3.10.6 torch-1.13.1+cu117 CUDA:0 (NVIDIA RTX A4500, 20470MiB)
YOLOv8m-seg summary (fused): 245 layers, 27268704 parameters, 0 gradients, 110.2 GFLOPs

image 1/1 /usr/local/lib/python3.10/dist-packages/ultralytics/assets/bus.jpg: 640x480 4 persons, 1 bus, 1 tie, 46.3ms
Speed: 0.3ms preprocess, 46.3ms inference, 2.6ms postprocess per image at shape (1, 3, 640, 640)
Ultralytics YOLOv8.0.45 🚀 Python-3.10.6 torch-1.13.1+cu117 CUDA:0 (NVIDIA RTX A4500, 20470MiB)
ERROR ❌️ Benchmark failure for PyTorch: [Errno 2] No such file or directory
Ultralytics YOLOv8.0.45 🚀 Python-3.10.6 torch-1.13.1+cu117 CUDA:0 (NVIDIA RTX A4500, 20470MiB)

PyTorch: starting from yolov8m-seg.pt with input shape (1, 3, 640, 640) BCHW and output shape(s) ((1, 116, 8400), (1, 32, 160, 160)) (52.4 MB)

TorchScript: starting export with torch 1.13.1+cu117...
TorchScript: export success ✅ 0.8s, saved as yolov8m-seg.torchscript (104.6 MB)

Export complete (0.9s)
Results saved to /workdir
Predict:         yolo predict task=segment model=yolov8m-seg.torchscript imgsz=640 
Validate:        yolo val task=segment model=yolov8m-seg.torchscript imgsz=640 data=coco.yaml 
Visualize:       https://netron.app/
Ultralytics YOLOv8.0.45 🚀 Python-3.10.6 torch-1.13.1+cu117 CUDA:0 (NVIDIA RTX A4500, 20470MiB)
Loading yolov8m-seg.torchscript for TorchScript inference...

image 1/1 /usr/local/lib/python3.10/dist-packages/ultralytics/assets/bus.jpg: 640x640 4 persons, 1 bus, 1 tie, 7.1ms
Speed: 0.4ms preprocess, 7.1ms inference, 1.7ms postprocess per image at shape (1, 3, 640, 640)
Ultralytics YOLOv8.0.45 🚀 Python-3.10.6 torch-1.13.1+cu117 CUDA:0 (NVIDIA RTX A4500, 20470MiB)
Loading yolov8m-seg.torchscript for TorchScript inference...
ERROR ❌️ Benchmark failure for TorchScript: [Errno 2] No such file or directory
Ultralytics YOLOv8.0.45 🚀 Python-3.10.6 torch-1.13.1+cu117 CUDA:0 (NVIDIA RTX A4500, 20470MiB)

PyTorch: starting from yolov8m-seg.pt with input shape (1, 3, 640, 640) BCHW and output shape(s) ((1, 116, 8400), (1, 32, 160, 160)) (52.4 MB)

ONNX: starting export with onnx 1.13.0...
ONNX: export success ✅ 1.1s, saved as yolov8m-seg.onnx (52.2 MB)

Export complete (1.1s)
Results saved to /workdir
Predict:         yolo predict task=segment model=yolov8m-seg.onnx imgsz=640 
Validate:        yolo val task=segment model=yolov8m-seg.onnx imgsz=640 data=coco.yaml 
Visualize:       https://netron.app/
Ultralytics YOLOv8.0.45 🚀 Python-3.10.6 torch-1.13.1+cu117 CUDA:0 (NVIDIA RTX A4500, 20470MiB)
Loading yolov8m-seg.onnx for ONNX Runtime inference...

image 1/1 /usr/local/lib/python3.10/dist-packages/ultralytics/assets/bus.jpg: 640x640 4 persons, 1 bus, 1 tie, 287.9ms
Speed: 0.4ms preprocess, 287.9ms inference, 4.0ms postprocess per image at shape (1, 3, 640, 640)
Ultralytics YOLOv8.0.45 🚀 Python-3.10.6 torch-1.13.1+cu117 CUDA:0 (NVIDIA RTX A4500, 20470MiB)
Loading yolov8m-seg.onnx for ONNX Runtime inference...
Forcing batch=1 square inference (1,3,640,640) for non-PyTorch models
ERROR ❌️ Benchmark failure for ONNX: [Errno 2] No such file or directory
ERROR ❌️ Benchmark failure for OpenVINO: inference not supported on GPU
Ultralytics YOLOv8.0.45 🚀 Python-3.10.6 torch-1.13.1+cu117 CUDA:0 (NVIDIA RTX A4500, 20470MiB)

PyTorch: starting from yolov8m-seg.pt with input shape (1, 3, 640, 640) BCHW and output shape(s) ((1, 116, 8400), (1, 32, 160, 160)) (52.4 MB)

ONNX: starting export with onnx 1.13.0...
ONNX: simplifying with onnxsim 0.4.17...
ONNX: export success ✅ 1.1s, saved as yolov8m-seg.onnx (52.2 MB)

TensorRT: starting export with TensorRT 8.5.3.1...
[02/25/2023-11:22:15] [TRT] [I] [MemUsageChange] Init CUDA: CPU +315, GPU +0, now: CPU 3706, GPU 3035 (MiB)
[02/25/2023-11:22:16] [TRT] [I] [MemUsageChange] Init builder kernel library: CPU +443, GPU +116, now: CPU 4204, GPU 3151 (MiB)
[02/25/2023-11:22:16] [TRT] [W] CUDA lazy loading is not enabled. Enabling it can significantly reduce device memory usage. See `CUDA_MODULE_LOADING` in https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#env-vars
[02/25/2023-11:22:16] [TRT] [I] ----------------------------------------------------------------
[02/25/2023-11:22:16] [TRT] [I] Input filename:   yolov8m-seg.onnx
[02/25/2023-11:22:16] [TRT] [I] ONNX IR version:  0.0.8
[02/25/2023-11:22:16] [TRT] [I] Opset version:    17
[02/25/2023-11:22:16] [TRT] [I] Producer name:    pytorch
[02/25/2023-11:22:16] [TRT] [I] Producer version: 1.13.1
[02/25/2023-11:22:16] [TRT] [I] Domain:           
[02/25/2023-11:22:16] [TRT] [I] Model version:    0
[02/25/2023-11:22:16] [TRT] [I] Doc string:       
[02/25/2023-11:22:16] [TRT] [I] ----------------------------------------------------------------
[02/25/2023-11:22:16] [TRT] [W] onnx2trt_utils.cpp:377: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
TensorRT: input "images" with shape(1, 3, 640, 640) DataType.HALF
TensorRT: output "output0" with shape(1, 116, 8400) DataType.HALF
TensorRT: output "output1" with shape(1, 32, 160, 160) DataType.HALF
TensorRT: building FP16 engine as yolov8m-seg.engine
[02/25/2023-11:22:16] [TRT] [I] [MemUsageChange] Init cuBLAS/cuBLASLt: CPU +0, GPU +32, now: CPU 4265, GPU 3183 (MiB)
[02/25/2023-11:22:16] [TRT] [I] [MemUsageChange] Init cuDNN: CPU +0, GPU +34, now: CPU 4265, GPU 3217 (MiB)
[02/25/2023-11:22:16] [TRT] [W] TensorRT was linked against cuDNN 8.6.0 but loaded cuDNN 8.5.0
[02/25/2023-11:22:16] [TRT] [I] Local timing cache in use. Profiling results in this builder pass will not be stored.
[02/25/2023-11:23:54] [TRT] [I] Some tactics do not have sufficient workspace memory to run. Increasing workspace size will enable more tactics, please check verbose output for requested sizes.
[02/25/2023-11:27:01] [TRT] [I] Total Activation Memory: 4548586496
[02/25/2023-11:27:01] [TRT] [I] Detected 1 inputs and 5 output network tensors.
[02/25/2023-11:27:01] [TRT] [I] Total Host Persistent Memory: 306560
[02/25/2023-11:27:01] [TRT] [I] Total Device Persistent Memory: 820224
[02/25/2023-11:27:01] [TRT] [I] Total Scratch Memory: 618496
[02/25/2023-11:27:01] [TRT] [I] [MemUsageStats] Peak memory usage of TRT CPU/GPU memory allocators: CPU 29 MiB, GPU 3793 MiB
[02/25/2023-11:27:01] [TRT] [I] [BlockAssignment] Started assigning block shifts. This will take 163 steps to complete.
[02/25/2023-11:27:01] [TRT] [I] [BlockAssignment] Algorithm ShiftNTopDown took 11.726ms to assign 11 blocks to 163 nodes requiring 51456512 bytes.
[02/25/2023-11:27:01] [TRT] [I] Total Activation Memory: 51456512
[02/25/2023-11:27:01] [TRT] [I] [MemUsageChange] Init cuDNN: CPU +0, GPU +32, now: CPU 4305, GPU 3329 (MiB)
[02/25/2023-11:27:01] [TRT] [W] TensorRT was linked against cuDNN 8.6.0 but loaded cuDNN 8.5.0
[02/25/2023-11:27:01] [TRT] [I] [MemUsageChange] TensorRT-managed allocation in building engine: CPU +7, GPU +53, now: CPU 7, GPU 53 (MiB)
TensorRT: export success ✅ 287.9s, saved as yolov8m-seg.engine (55.3 MB)

Export complete (288.0s)
Results saved to /workdir
Predict:         yolo predict task=segment model=yolov8m-seg.engine imgsz=640 
Validate:        yolo val task=segment model=yolov8m-seg.engine imgsz=640 data=coco.yaml 
Visualize:       https://netron.app/
Ultralytics YOLOv8.0.45 🚀 Python-3.10.6 torch-1.13.1+cu117 CUDA:0 (NVIDIA RTX A4500, 20470MiB)
Loading yolov8m-seg.engine for TensorRT inference...
[02/25/2023-11:27:01] [TRT] [I] The logger passed into createInferRuntime differs from one already provided for an existing builder, runtime, or refitter. Uses of the global logger, returned by nvinfer1::getLogger(), will return the existing value.

[02/25/2023-11:27:01] [TRT] [I] Loaded engine size: 55 MiB
[02/25/2023-11:27:01] [TRT] [I] [MemUsageChange] Init cuDNN: CPU +0, GPU +32, now: CPU 4293, GPU 3247 (MiB)
[02/25/2023-11:27:01] [TRT] [W] TensorRT was linked against cuDNN 8.6.0 but loaded cuDNN 8.5.0
[02/25/2023-11:27:01] [TRT] [I] [MemUsageChange] TensorRT-managed allocation in engine deserialization: CPU +0, GPU +52, now: CPU 0, GPU 52 (MiB)
[02/25/2023-11:27:01] [TRT] [I] [MemUsageChange] Init cuDNN: CPU +0, GPU +32, now: CPU 4238, GPU 3247 (MiB)
[02/25/2023-11:27:01] [TRT] [W] TensorRT was linked against cuDNN 8.6.0 but loaded cuDNN 8.5.0
[02/25/2023-11:27:01] [TRT] [I] [MemUsageChange] TensorRT-managed allocation in IExecutionContext creation: CPU +0, GPU +50, now: CPU 0, GPU 102 (MiB)
[02/25/2023-11:27:01] [TRT] [W] CUDA lazy loading is not enabled. Enabling it can significantly reduce device memory usage. See `CUDA_MODULE_LOADING` in https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#env-vars

image 1/1 /usr/local/lib/python3.10/dist-packages/ultralytics/assets/bus.jpg: 640x640 4 persons, 1 bus, 1 tie, 41.0ms
Speed: 1.5ms preprocess, 41.0ms inference, 5.1ms postprocess per image at shape (1, 3, 640, 640)
Ultralytics YOLOv8.0.45 🚀 Python-3.10.6 torch-1.13.1+cu117 CUDA:0 (NVIDIA RTX A4500, 20470MiB)
Loading yolov8m-seg.engine for TensorRT inference...
[02/25/2023-11:27:02] [TRT] [I] The logger passed into createInferRuntime differs from one already provided for an existing builder, runtime, or refitter. Uses of the global logger, returned by nvinfer1::getLogger(), will return the existing value.

[02/25/2023-11:27:02] [TRT] [I] Loaded engine size: 55 MiB
[02/25/2023-11:27:02] [TRT] [I] [MemUsageChange] Init cuDNN: CPU +0, GPU +32, now: CPU 4308, GPU 3385 (MiB)
[02/25/2023-11:27:02] [TRT] [W] TensorRT was linked against cuDNN 8.6.0 but loaded cuDNN 8.5.0
[02/25/2023-11:27:02] [TRT] [I] [MemUsageChange] TensorRT-managed allocation in engine deserialization: CPU +0, GPU +53, now: CPU 0, GPU 155 (MiB)
[02/25/2023-11:27:02] [TRT] [I] [MemUsageChange] Init cuDNN: CPU +0, GPU +32, now: CPU 4253, GPU 3387 (MiB)
[02/25/2023-11:27:02] [TRT] [W] TensorRT was linked against cuDNN 8.6.0 but loaded cuDNN 8.5.0
[02/25/2023-11:27:02] [TRT] [I] [MemUsageChange] TensorRT-managed allocation in IExecutionContext creation: CPU +1, GPU +50, now: CPU 1, GPU 205 (MiB)
[02/25/2023-11:27:02] [TRT] [W] CUDA lazy loading is not enabled. Enabling it can significantly reduce device memory usage. See `CUDA_MODULE_LOADING` in https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#env-vars
ERROR ❌️ Benchmark failure for TensorRT: [Errno 2] No such file or directory
ERROR ❌️ Benchmark failure for CoreML: inference only supported on macOS>=10.13
Ultralytics YOLOv8.0.45 🚀 Python-3.10.6 torch-1.13.1+cu117 CUDA:0 (NVIDIA RTX A4500, 20470MiB)

PyTorch: starting from yolov8m-seg.pt with input shape (1, 3, 640, 640) BCHW and output shape(s) ((1, 116, 8400), (1, 32, 160, 160)) (52.4 MB)
WARNING ⚠️ YOLOv8 TensorFlow export is still under development. Please consider contributing to the effort if you have TF expertise. Thank you!
2023-02-25 11:27:02.377011: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 AVX_VNNI FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-02-25 11:27:02.482968: I tensorflow/core/util/port.cc:104] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2023-02-25 11:27:02.826727: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer.so.7'; dlerror: libnvinfer.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/lib/python3.10/dist-packages/cv2/../../lib64:/.singularity.d/libs
2023-02-25 11:27:02.826800: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer_plugin.so.7'; dlerror: libnvinfer_plugin.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/lib/python3.10/dist-packages/cv2/../../lib64:/.singularity.d/libs
2023-02-25 11:27:02.826817: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.

TensorFlow SavedModel: starting export with tensorflow 2.11.0...

ONNX: starting export with onnx 1.13.0...
ONNX: simplifying with onnxsim 0.4.17...
ONNX: export success ✅ 2.2s, saved as yolov8m-seg.onnx (104.2 MB)

TensorFlow SavedModel: running onnx2tf -i yolov8m-seg.onnx -o yolov8m-seg_saved_model --non_verbose 
WARNING:absl:Please consider providing the trackable_obj argument in the from_concrete_functions. Providing without the trackable_obj argument is deprecated and it will use the deprecated conversion path.
2023-02-25 11:27:17.289934: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:967] could not open file to read NUMA node: /sys/bus/pci/devices/0000:01:00.0/numa_node
Your kernel may have been built without NUMA support.
2023-02-25 11:27:17.290103: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcufft.so.10'; dlerror: libcufft.so.10: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/lib/python3.10/dist-packages/cv2/../../lib64:/.singularity.d/libs
2023-02-25 11:27:17.290199: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcurand.so.10'; dlerror: libcurand.so.10: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/lib/python3.10/dist-packages/cv2/../../lib64:/.singularity.d/libs
2023-02-25 11:27:17.290247: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcusolver.so.11'; dlerror: libcusolver.so.11: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/lib/python3.10/dist-packages/cv2/../../lib64:/.singularity.d/libs
2023-02-25 11:27:17.290405: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcusparse.so.11'; dlerror: libcusparse.so.11: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/lib/python3.10/dist-packages/cv2/../../lib64:/.singularity.d/libs
2023-02-25 11:27:17.290438: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1934] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...
2023-02-25 11:27:17.290609: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 AVX_VNNI FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
TensorFlow SavedModel: export success ✅ 15.0s, saved as yolov8m-seg_saved_model (260.7 MB)

Export complete (15.3s)
Results saved to /workdir
Predict:         yolo predict task=segment model=yolov8m-seg_saved_model imgsz=640 
Validate:        yolo val task=segment model=yolov8m-seg_saved_model imgsz=640 data=coco.yaml 
Visualize:       https://netron.app/
Ultralytics YOLOv8.0.45 🚀 Python-3.10.6 torch-1.13.1+cu117 CUDA:0 (NVIDIA RTX A4500, 20470MiB)
Loading yolov8m-seg_saved_model for TensorFlow SavedModel inference...

image 1/1 /usr/local/lib/python3.10/dist-packages/ultralytics/assets/bus.jpg: 640x640 23 persons, 3 buss, 1 tie, 172.6ms
Speed: 1.9ms preprocess, 172.6ms inference, 11.3ms postprocess per image at shape (1, 3, 640, 640)
Ultralytics YOLOv8.0.45 🚀 Python-3.10.6 torch-1.13.1+cu117 CUDA:0 (NVIDIA RTX A4500, 20470MiB)
Loading yolov8m-seg_saved_model for TensorFlow SavedModel inference...
Forcing batch=1 square inference (1,3,640,640) for non-PyTorch models
ERROR ❌️ Benchmark failure for TensorFlow SavedModel: [Errno 2] No such file or directory
Ultralytics YOLOv8.0.45 🚀 Python-3.10.6 torch-1.13.1+cu117 CUDA:0 (NVIDIA RTX A4500, 20470MiB)

PyTorch: starting from yolov8m-seg.pt with input shape (1, 3, 640, 640) BCHW and output shape(s) ((1, 116, 8400), (1, 32, 160, 160)) (52.4 MB)
WARNING ⚠️ YOLOv8 TensorFlow export is still under development. Please consider contributing to the effort if you have TF expertise. Thank you!

TensorFlow SavedModel: starting export with tensorflow 2.11.0...

ONNX: starting export with onnx 1.13.0...
ONNX: simplifying with onnxsim 0.4.17...
ONNX: export success ✅ 1.7s, saved as yolov8m-seg.onnx (104.2 MB)

TensorFlow SavedModel: running onnx2tf -i yolov8m-seg.onnx -o yolov8m-seg_saved_model --non_verbose 
WARNING:absl:Please consider providing the trackable_obj argument in the from_concrete_functions. Providing without the trackable_obj argument is deprecated and it will use the deprecated conversion path.
TensorFlow SavedModel: export success ✅ 13.9s, saved as yolov8m-seg_saved_model (260.7 MB)

TensorFlow GraphDef: starting export with tensorflow 2.11.0...
WARNING:tensorflow:From /usr/local/lib/python3.10/dist-packages/tensorflow/python/autograph/pyct/static_analysis/liveness.py:83: Analyzer.lamba_check (from tensorflow.python.autograph.pyct.static_analysis.liveness) is deprecated and will be removed after 2023-09-23.
Instructions for updating:
Lambda fuctions will be no more assumed to be used in the statement where they are used, or at least in the same block. https://github.com/tensorflow/tensorflow/issues/56089
2023-02-25 11:27:32.357663: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:967] could not open file to read NUMA node: /sys/bus/pci/devices/0000:01:00.0/numa_node
Your kernel may have been built without NUMA support.
2023-02-25 11:27:32.357744: I tensorflow/core/grappler/devices.cc:66] Number of eligible GPUs (core count >= 8, compute capability >= 0.0): 1
2023-02-25 11:27:32.357833: I tensorflow/core/grappler/clusters/single_machine.cc:358] Starting new session
2023-02-25 11:27:32.358034: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:967] could not open file to read NUMA node: /sys/bus/pci/devices/0000:01:00.0/numa_node
Your kernel may have been built without NUMA support.
2023-02-25 11:27:32.358061: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1934] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...
TensorFlow GraphDef: export success ✅ 0.8s, saved as yolov8m-seg.pb (104.3 MB)

Export complete (14.9s)
Results saved to /workdir
Predict:         yolo predict task=segment model=yolov8m-seg.pb imgsz=640 data=coco.yaml
Validate:        yolo val task=segment model=yolov8m-seg.pb imgsz=640 data=coco.yaml 
Visualize:       https://netron.app/
Ultralytics YOLOv8.0.45 🚀 Python-3.10.6 torch-1.13.1+cu117 CUDA:0 (NVIDIA RTX A4500, 20470MiB)
Loading yolov8m-seg.pb for TensorFlow GraphDef inference...

image 1/1 /usr/local/lib/python3.10/dist-packages/ultralytics/assets/bus.jpg: 640x640 23 class0s, 3 class5s, 1 class27, 147.7ms
Speed: 1.9ms preprocess, 147.7ms inference, 11.3ms postprocess per image at shape (1, 3, 640, 640)
Ultralytics YOLOv8.0.45 🚀 Python-3.10.6 torch-1.13.1+cu117 CUDA:0 (NVIDIA RTX A4500, 20470MiB)
Loading yolov8m-seg.pb for TensorFlow GraphDef inference...
Forcing batch=1 square inference (1,3,640,640) for non-PyTorch models
ERROR ❌️ Benchmark failure for TensorFlow GraphDef: [Errno 2] No such file or directory
ERROR ❌️ Benchmark failure for TensorFlow Lite: inference not supported on GPU
ERROR ❌️ Benchmark failure for TensorFlow Edge TPU: inference not supported
ERROR ❌️ Benchmark failure for TensorFlow.js: inference not supported
Ultralytics YOLOv8.0.45 🚀 Python-3.10.6 torch-1.13.1+cu117 CUDA:0 (NVIDIA RTX A4500, 20470MiB)

PyTorch: starting from yolov8m-seg.pt with input shape (1, 3, 640, 640) BCHW and output shape(s) ((1, 116, 8400), (1, 32, 160, 160)) (52.4 MB)
Error: Can not import paddle core while this file exists: /usr/local/lib/python3.10/dist-packages/paddle/fluid/libpaddle.so
PaddlePaddle: export failure ❌ 0.0s: libcudart.so.10.2: cannot open shared object file: No such file or directory
ERROR ❌️ Benchmark failure for PaddlePaddle: expected str, bytes or os.PathLike object, not list


Ultralytics YOLOv8.0.45 🚀 Python-3.10.6 torch-1.13.1+cu117 CUDA:0 (NVIDIA RTX A4500, 20470MiB)
Setup complete ✅ (24 CPUs, 31.2 GB RAM, 130.2/1006.9 GB disk)

Benchmarks complete for yolov8m-seg.pt on coco128-seg.yaml at imgsz=640 (328.66s)
                   Format Status❔ Size (MB) metrics/mAP50-95(M) Inference time (ms/im)
0                 PyTorch       ❌      None                None                   None
1             TorchScript       ❌      None                None                   None
2                    ONNX       ❌      None                None                   None
3                OpenVINO       ❌      None                None                   None
4                TensorRT       ❌      None                None                   None
5                  CoreML       ❌      None                None                   None
6   TensorFlow SavedModel       ❌      None                None                   None
7     TensorFlow GraphDef       ❌      None                None                   None
8         TensorFlow Lite       ❌      None                None                   None
9     TensorFlow Edge TPU       ❌      None                None                   None
10          TensorFlow.js       ❌      None                None                   None
11           PaddlePaddle       ❌      None                None                   None
```

**Comment by glenn-jocher** [comment]
@titq2022 your command works correctly for me. I'm unable to produce any issues with it, i.e.:

<img width="1193" alt="Screenshot 2023-02-24 at 19 08 54" src="https://user-images.githubusercontent.com/26833433/221333097-7c6308a5-4dbf-4841-aa59-b0c69f6727f8.png">

**Comment by titq2022** [comment]
Thanks glenn-jocher.

I tried the tutorial.
This is what it looks like when it works correctly.
I would like to get it working in my environment.

Inside benchmarks.py, predict is executed after export, then val is executed. 
In my environment, predict works, but I had not tried val, so I tried it.
**my code**
`!yolo segment val model=/workdir/yolov8m-seg.pt data=/workdir/datasets/coco128-seg.yaml`
**trace**
```
Ultralytics YOLOv8.0.45 🚀 Python-3.10.6 torch-1.13.1+cu117 CUDA:0 (NVIDIA RTX A4500, 20470MiB)
YOLOv8m-seg summary (fused): 245 layers, 27268704 parameters, 0 gradients, 110.2 GFLOPs
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/ultralytics/yolo/data/dataset.py", line 100, in get_labels
    cache, exists = np.load(str(cache_path), allow_pickle=True).item(), True  # load dict
  File "/usr/local/lib/python3.10/dist-packages/numpy/lib/npyio.py", line 390, in load
    fid = stack.enter_context(open(os_fspath(file), "rb"))
FileNotFoundError: [Errno 2] No such file or directory: '/workdir/datasets/coco128-seg/labels/train2017.cache'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/yolo", line 8, in <module>
    sys.exit(entrypoint())
  File "/usr/local/lib/python3.10/dist-packages/ultralytics/yolo/cfg/__init__.py", line 304, in entrypoint
    getattr(model, mode)(**overrides)  # default args from model
  File "/usr/local/lib/python3.10/dist-packages/torch/autograd/grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/ultralytics/yolo/engine/model.py", line 241, in val
    validator(model=self.model)
  File "/usr/local/lib/python3.10/dist-packages/torch/autograd/grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/ultralytics/yolo/engine/validator.py", line 136, in __call__
    self.dataloader = self.dataloader or self.get_dataloader(self.data.get(self.args.split), self.args.batch)
  File "/usr/local/lib/python3.10/dist-packages/ultralytics/yolo/v8/detect/val.py", line 180, in get_dataloader
    build_dataloader(self.args, batch_size, img_path=dataset_path, stride=gs, names=self.data['names'],
  File "/usr/local/lib/python3.10/dist-packages/ultralytics/yolo/data/build.py", line 71, in build_dataloader
    dataset = YOLODataset(
  File "/usr/local/lib/python3.10/dist-packages/ultralytics/yolo/data/dataset.py", line 44, in __init__
    super().__init__(img_path, imgsz, cache, augment, hyp, prefix, rect, batch_size, stride, pad, single_cls)
  File "/usr/local/lib/python3.10/dist-packages/ultralytics/yolo/data/base.py", line 49, in __init__
    self.labels = self.get_labels()
  File "/usr/local/lib/python3.10/dist-packages/ultralytics/yolo/data/dataset.py", line 104, in get_labels
    cache, exists = self.cache_labels(cache_path), False  # run cache ops
  File "/usr/local/lib/python3.10/dist-packages/ultralytics/yolo/data/dataset.py", line 52, in cache_labels
    with ThreadPool(NUM_THREADS) as pool:
  File "/usr/lib/python3.10/multiprocessing/pool.py", line 930, in __init__
    Pool.__init__(self, processes, initializer, initargs)
  File "/usr/lib/python3.10/multiprocessing/pool.py", line 196, in __init__
    self._change_notifier = self._ctx.SimpleQueue()
  File "/usr/lib/python3.10/multiprocessing/context.py", line 113, in SimpleQueue
    return SimpleQueue(ctx=self.get_context())
  File "/usr/lib/python3.10/multiprocessing/queues.py", line 341, in __init__
    self._rlock = ctx.Lock()
  File "/usr/lib/python3.10/multiprocessing/context.py", line 68, in Lock
    return Lock(ctx=self.get_context())
  File "/usr/lib/python3.10/multiprocessing/synchronize.py", line 162, in __init__
    SemLock.__init__(self, SEMAPHORE, 1, 1, ctx=ctx)
  File "/usr/lib/python3.10/multiprocessing/synchronize.py", line 57, in __init__
    sl = self._semlock = _multiprocessing.SemLock(
FileNotFoundError: [Errno 2] No such file or directory
```

`FileNotFoundError: [Errno 2] No such file or directory: '/workdir/datasets/coco128-seg/labels/train2017.cache'`
It seems to be caused by the` train2017.cache` not existing, is the .cache file automatically created when the val is run? 
Have I left out any steps?

**Comment by glenn-jocher** [comment]
@titq2022 no additional steps are necessary, just pip install and run a training command. If your commands work in Colab but not locally then your files or environment may be the cause. If your code or coco128-seg dataset is modified or out of date just delete it and install again and rerun.

**Comment by titq2022** [comment]
@glenn-jocher I tried it with a lower version of Python. It worked until about halfway through, but gave different errors. 
It seemed difficult for me to solve, so I decided to use the ultralytics docker image.

Ultralytics YOLOv8.0.47 🚀 Python-3.10.8 torch-1.13.1 CUDA:0 (NVIDIA RTX A4500, 20470MiB)
![image](https://user-images.githubusercontent.com/115346408/221401073-1f4bc3c1-918a-4159-829d-2632593d8c46.png)
